gql/dsl.py

```
@@ -61,49 +65,36 @@ def ast_from_value(value: Any, type_: GraphQLInputType) -> Optional[ValueNode]:
     if isinstance(value, DSLVariable):
         return value.set_type(type_).ast_variable
 
-    if is_non_null_type(type_):
-        type_ = cast(GraphQLNonNull, type_)
-        ast_value = ast_from_value(value, type_.of_type)
-        if isinstance(ast_value, NullValueNode):
-            return None
-        return ast_value
+    if isinstance(type_, GraphQLNonNull):
+        return ast_from_value(value, type_.of_type)
 
-    # only explicit None, not Undefined or NaN
-    if value is None:
-        return NullValueNode()
-
-    # undefined
     if value is Undefined:
         return None
 
-    # Convert Python list to GraphQL list. If the GraphQLType is a list, but the value
-    # is not a list, convert the value using the list's item type.
-    if is_list_type(type_):
-        type_ = cast(GraphQLList, type_)
-        item_type = type_.of_type
-        if isinstance(value, Iterable) and not isinstance(value, str):
-            maybe_value_nodes = (ast_from_value(item, item_type) for item in value)
-            value_nodes = filter(None, maybe_value_nodes)
-            return ListValueNode(values=FrozenList(value_nodes))
-        return ast_from_value(value, item_type)
-
-    # Populate the fields of the input object by creating ASTs from each value in the
-    # Python dict according to the fields in the input type.
-    if is_input_object_type(type_):
+    if isinstance(value, list):
+        item_type = type_.of_type if isinstance(type_, GraphQLList) else None
+        return ListValueNode([ast_from_value(item, item_type) for item in value])
```

question: If value is a list, but `type_` is `GraphQLScalarType` then should it fail instead of returning `ListValueNode`?

---

This is a verbatim copy from
https://github.com/graphql-python/graphql-core-legacy/blob/master/graphql/utils/ast_from_value.py

so I wouldn't change that.